lib/http/request/writer.rb

```diff
@@ -60,22 +60,33 @@ def join_headers
         @request_header.join(CRLF) + CRLF * 2
       end
 
+      # Writes HTTP request data into the socket.
       def send_request
-        # It's important to send the request in a single write call when
-        # possible in order to play nicely with Nagle's algorithm. Making
-        # two writes in a row triggers a pathological case where Nagle is
-        # expecting a third write that never happens.
+        each_chunk do |chunk|
+          write(chunk)
+        end
+      rescue Errno::EPIPE # rubocop:disable Lint/HandleExceptions
+        # server doesn't need any more data
```

Comment:
IMO it worth to explicitly add `nil` at the end of `rescue` block and remove rubocop disable.

Reply:
Oh. It was discussed previously... Mmm... I still think we should just return `nil` from the `rescue`. Also body can be shortened:

``` ruby
def send_request
  each_chunk { |chunk| write chunk }
rescue Errno::EPIPE
  # no more data
  nil
end
```

Reply:
On one hand I think that `Lint/HandleExceptions` is pretty useless and enforces developer to write more than needed. On the other hand I prefer to see:

``` ruby
begin
  # ...
rescue
  nil
end
```

than

``` ruby
begin
  # ...
rescue
end
```

IMO it's more readable that way. Easier to understand that `nil` is result in case of error.

Reply:
I know that one might argue that we don't require explicit nil in case of `return if something`, but it's a bit different - it's one-liner. And reading it is pretty straight-forward to me. And when we explicitly return from non-one-liner code we add `nil` to return:

``` ruby
if something
  # ...
  return nil # <-- explicit `nil` here looks better than bare `return`
end
```

Reply:
Agreed, updated.